File: python/003.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import sys

def factor(n):
    i = 2
    factors = []
    while i * i <= n:
        while n % i == 0:
            n /= i
            factors.append(i)
        i += 1
    if n > 1:
        factors.append(int(n))
    return factors

def p3(n):
    return max(factor(n))

def main():
    print(p3(600851475143))

if __name__ == '__main__':
    main()

File: elixir/003.exs
#!/usr/bin/env elixir

# https://projecteuler.net/problem=3

defmodule P3 do
	def factor(n) do
		factor(n, 2, [])
	end
	def factor(1, _, factors) do
		factors
	end
	def factor(n, m, factors) when rem(n, m) == 0 do
		factor(div(n, m), m, [ m | factors ]);
	end
	def factor(n, m, factors) do
		factor(n, m + 1, factors)
	end
end

P3.factor(600851475143)
|> Enum.max
|> IO.inspect
